java/client/src/main/java/glide/connectors/handlers/ChannelHandler.java

```diff
@@ -58,7 +58,7 @@ public ChannelHandler(
                         .channel(domainSocketChannelClass)
                         .handler(channelInitializer)
                         .connect(domainSocketAddress)
-                        // TODO call here .sync() if needed or remove this comment
+                        .syncUninterruptibly()
```

Comment:
interruption happens, where will you handle it?

Reply:
https://github.com/aws/glide-for-redis/blob/7ad8e5458f8f4a35f2fc719053628570e2932821/java/client/src/main/java/glide/api/RedisClient.java#L28
We have a place to handle this in future. Also it is tracked in our TODO task.

Reply:
Why not `.sync()`?

Reply:
`.sync()` throws `InterrupedException` which we can rethrow down to `RedisClient.CreateClient` and further to the user or suppress as `syncUninterrupibly` does.

Reply:
Which means we can mark the Future response appropriately.  
If we just suppress the call, the user won't know.

Reply:
This happens before Future is created. We still can create it and terminate it with an exception.

Reply:
On connection?  Sure, we can interrupt the future.

Reply:
> interrput the future

sounds like casting an apocalypse

See fix in 59c0b81a